src/transformers/utils/import_utils.py

```diff
@@ -443,7 +443,29 @@ def is_apex_available():
 
 
 def is_ipex_available():
-    return importlib.util.find_spec("intel_extension_for_pytorch") is not None
+    def get_major_and_minor_from_version(version):
+        import re
+
+        matches = re.match("(\d+\.\d+).*", version)
+        if matches and len(matches.groups()) == 1:
+            return matches.group(1)
+
+    if not is_torch_available() or importlib.util.find_spec("intel_extension_for_pytorch") is None:
+        return False
+    _ipex_version = ""
+    try:
+        _ipex_version = importlib_metadata.version("intel_extension_for_pytorch")
+    except importlib_metadata.PackageNotFoundError:
+        return False
+    torch_major_and_minor = get_major_and_minor_from_version(_torch_version)
+    ipex_major_and_minor = get_major_and_minor_from_version(_ipex_version)
```

Comment:
Please use `version.parse` here (`version` is imported from `packaging` in this module), then rely on the `.minor` and `.major` attributes of the objects returned.

Reply:
something as simple as:
```
if parse(torch.__version__) != parse(ipex.__version__):
```
 :)

I assume `ipex.__version__` exists.

Reply:
Thanks for the suggestion! Here have made changes to refine the version parsing and comparing accordingly.